packages/dev/loaders/src/glTF/2.0/Extensions/KHR_animation_pointer.ts

```diff
@@ -0,0 +1,343 @@
+     * @param animationTargetOverride The babylon animation channel target override property. My be null.
+     * @returns A void promise when the channel load is complete
+     */
+    public _loadAnimationChannelAsync(
+        context: string,
+        animationContext: string,
+        animation: IAnimation,
+        channel: IAnimationChannel,
+        animationTargetOverride: Nullable<IAnimatable> = null
+    ): Promise<void> {
+        if (channel.target.path != AnimationChannelTargetPath.POINTER) {
+            throw new Error(`${context}/target/path: Invalid value (${channel.target.path})`);
+        }
+
+        if (channel.target.node != undefined) {
+            // According to KHR_animation_pointer specification
+            // If this extension is used, the animation.channel.target.node must not be set.
+            // Because the node isn’t defined, the channel is ignored and not animated due to the specification.
+            return Promise.resolve();
+        }
+
+        const pointer = channel.target.extensions?.KHR_animation_pointer?.pointer;
+        if (!pointer) {
+            throw new Error(`${context}/target/extensions.KHR_animation_pointer.pointer MUST be set.`);
+        }
+
+        const sampler = ArrayItem.Get(`${context}/sampler`, animation.samplers, channel.sampler);
+
+        return this._loadAnimationSamplerAsync(`${context}/samplers/${channel.sampler}`, sampler).then((data) => {
+            // this is where we process the pointer.
+            const animationTarget = this._parseAnimationPointer(`${context}/extensions/${this.name}/pointer`, pointer);
+
+            if (animationTarget) {
+                // build the keys
+                // build the animations into the group
+                const babylonAnimationGroup = animation._babylonAnimationGroup;
+                if (babylonAnimationGroup) {
+                    const outputAccessor = ArrayItem.Get(`${context}/output`, this._loader.gltf.accessors, sampler.output);
+                    // stride is the size of each element stored into the output buffer.
+                    const stride = animationTarget.stride ?? KHR_animation_pointer._GetAnimationOutputStride(outputAccessor.type);
+                    const fps = this._loader.parent.targetFps;
+
+                    // we extract the corresponding values from the readed value.
+                    // the reason for that is one GLTF value may be dispatched to several Babylon properties
+                    // one of example is baseColorFactor which is a Color4 under GLTF and dispatched to
+                    // - albedoColor as Color3(Color4.r,Color4.g,Color4.b)
+                    // - alpha as Color4.a
+                    for (const propertyInfo of animationTarget.properties) {
+                        // Ignore animations that have no animation valid targets.
+                        if (!propertyInfo.isValid(animationTarget.target)) {
+                            return Promise.resolve();

```

Comment:
nit: since you are already in a continuation, you just need to return.

```suggestion
                            return;
```

Reply:
wasn't know